packages/nimble-components/src/rich-text/editor/index.ts

```diff
@@ -335,6 +336,53 @@ export class RichTextEditor extends FoundationElement implements ErrorPattern {
 
     private createTiptapEditor(): Editor {
         const customLink = this.getCustomLinkExtension();
+        const validAbsoluteLinkRegex = /^https?:\/\//i;
+
+        /**
+         * @param htmlString contains the html string of the copied content. If the content is a link, the `htmlString` contains anchor tag and a href value.
+         * ProseMirror reference for `transformPastedHTML`: https://prosemirror.net/docs/ref/#view.EditorProps.transformPastedHTML
+         */
+        const transformPastedHTML = (htmlString: string): string => {
```

Comment:
What is the expected behavior for plain text URLs embedded within a larger paragraph? For example:
```
The URL https://www.ni.com takes you to NI's homepage.
```

I'm seeing that these URLs remain plan text unless you move the cursor to the end of the URL and type space/Enter.

Is that expected (or at least a known limitation)? I think my personal expectation would be for the editor to detect valid URLs within the entire pasted text. That seems to be the expectation in the reported bug too, since it is complaining about having to type space/enter after pasting a link (though it's unclear if that was a plain text link or a hyperlink). [Bug 2525827](https://dev.azure.com/ni/DevCentral/_workitems/edit/2525827): Rich text editor | Absolute links requires space or enter after pasting the link

Reply:
I think we need to push back on this whole direction. 
1. It seems like the whole thing is a hack that will be removed when full link support is added based on the comment "Pasting rules need not to be transformed when hyperlink support added".
2. It was not a trivial fix. The assumption was incorrect. Safely handling HTML has been difficult. First it was done unsafely and then I suggested an alternative that I think might be safe and asked to find a way to confirm that but did not receive a compelling confirmation. 
    
    The concern was not that upstream tiptap will safely handle pasted HTML upstream inside the editor. The concern is that we are safely doing it which deep diving into the editor does not confirm.
3. It seems to require duplicating work tiptap, etc are handling upstream (sanitization, parsing paste results, etc).
4. It is exposing us to a whole new domain I don't know how to deal with correctly. I see a whole bunch of logic added to handle nimble-anchor tags in the pasted html I do not understand and was not cited in comments. Turns out that copying and pasting a rendered nimble-anchor results in a html string chunk like the following:
    ```
    <html>
    <body>
    <!--StartFragment--><span style="color: rgb(0, 0, 0); font-family: &quot;Source Sans Pro&quot;, &quot;Source Sans Pro Fallback&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(244, 244, 244); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial; display: inline !important; float: none;"><span> </span>the<span> </span></span><nimble-anchor href="https://nimble.ni.dev" appearance="default" style="display: inline; box-sizing: border-box; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-variant-numeric: ; font-variant-east-asian: ; font-variant-alternates: ; font-variant-position: ; font-weight: 400; font-stretch: ; font-size: 14px; line-height: ; font-family: &quot;Source Sans Pro&quot;, &quot;Source Sans Pro Fallback&quot;; font-optical-sizing: ; font-kerning: ; font-feature-settings: ; font-variation-settings: ; color: rgb(0, 0, 0); letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(244, 244, 244); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><a class="control" part="control" href="https://nimble.ni.dev/" style="color: var(--ni-nimble-link-font-color); text-decoration: underline;"><span class="content" part="content"><slot>link</slot></span></a></nimble-anchor><span style="color: rgb(0, 0, 0); font-family: &quot;Source Sans Pro&quot;, &quot;Source Sans Pro Fallback&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(244, 244, 244); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial; display: inline !important; float: none;"><span> </span>to n</span><!--EndFragment-->
    </body>
    </html>
    ```
    
    Some takeaways: 
        - The copy and paste results in the internal shadow dom of the nimble-anchor being exposed in the html chunk and all the internal html like slots, etc. Which is kinda nuts to deal with.
        - Styling gets inlined in some way I don't understand
        - I have no idea how cross-browser consistent any of this is

Reply:
The expected behavior that we are trying to achieve here is if the URL is already a link when copied from the source, it should render as a link; if it is plain text, it can remain plain text. Here is the short GIF explaining this scenario copied from a viewer and pasting it in the editor. Even the AzDo Board comments work the same.
![copy-paste-link-in-paragraph](https://github.com/ni/nimble/assets/123377523/e92536c7-a441-4bc9-b5e6-d1a97251aaff)

Reply:
As discussed offline, this bug is needed for this cycle release in SLE and so we have to add this workaround. We also aligned on all the above points and captured different issues in respective issues for tracking the work in the future. So I am closing this for now.